how-tos/add-translations.md

```diff
@@ -0,0 +1,122 @@
+
+## Supporting Multiple Languages in a Reaction Client
+
+In browser code, initialize multi-language support like this, using latest `i18next` NPM packages:
+
+```js
+import i18next from "i18next";
+import i18nextBrowserLanguageDetector from "i18next-browser-languagedetector";
+import i18nextFetch from "i18next-fetch-backend";
+import i18nextMultiLoadBackendAdapter from "i18next-multiload-backend-adapter";
+import { initReactI18next } from "react-i18next";
+
+const configuredI18next = i18next
+  // https://github.com/i18next/i18next-browser-languageDetector
+  // Sets initial language to load based on `lng` query string
+  // with various fallbacks.
+  .use(i18nextBrowserLanguageDetector)
+  // https://github.com/perrin4869/i18next-fetch-backend
+  // This uses `fetch` to load resources from the backend based on `backend`
+  // config object below.
+  .use(i18nextMultiLoadBackendAdapter)
+  // pass the i18n instance to react-i18next.
+  .use(initReactI18next);
+
+/**
+ * @summary Initializes i18next
+ * @param {String} fallbackLng Language code to use if i18nextBrowserLanguageDetector fails
+ * @return {undefined}
+ */
+async function initializeI18n(fallbackLng) {
+  // Reaction does not have a predefined list of namespaces. Any API plugin can
+  // add any namespaces. So we must first get the list of namespaces from the API.
+  const namespaceResponse = await fetch(`${apiUrl}/locales/namespaces.json`);
+  const allTranslationNamespaces = await namespaceResponse.json();
+
+  await configuredI18next.init({
+    backend: {
+      backend: i18nextFetch,
+      backendOption: {
+        allowMultiLoading: true,
+        loadPath: `${apiUrl}/locales/resources.json?lng={{lng}}&ns={{ns}}`
+      }
+    },
+    debug: false,
+    detection: {
+      // We primarily set language according to `navigator.language`,
+      // which is supported in all modern browsers and can be changed
+      // in the browser settings. This is the same list that browsers
+      // send in the `Accept-Language` header.
+      //
+      // For ease of testing translations, we also support `lng`
+      // query string to override the browser setting.
+      order: ["querystring", "navigator"]
+    },
+    ns: allTranslationNamespaces,
+    defaultNS: "core", // reaction "core" is the default namespace
+    fallbackNS: allTranslationNamespaces,
+    fallbackLng
+  });
+}
+
+initializeI18n("en").catch((error) => {
+  console.error(error);
+})
+```
+
+This will show the UI in the user's preferred language, based on their browser setting. For example, in Chrome, go to chrome://settings/languages and whatever language is listed first in your preferences list will be the language used.
+
+To see the UI in a specific language, overriding your browser language, you can append `lng=<code>` in your query string. For example, `http://localhost:3000/?lng=en` or `http://localhost:3000/?lng=en-US` will show you English text even if your browser language is set to something else.
```

Comment:
@zenweasel i need to test this but at least change ports to 4000 so it refers to storefront